packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -198,12 +196,11 @@ export abstract class PipelineCloudFormationDeployAction extends PipelineCloudFo
   public readonly role: iam.Role;
 
   constructor(parent: cdk.Construct, id: string, props: PipelineCloudFormationDeployActionProps, configuration: any) {
-    const capabilities = props.fullPermissions && props.capabilities === undefined ? [CloudFormationCapabilities.NamedIAM] : props.capabilities;
-
+    const capabilities = props.adminPermissions && props.capabilities === undefined ? CloudFormationCapabilities.NamedIAM : props.capabilities;
     super(parent, id, props, {
       ...configuration,
-      // This must be a string, so flatten the list to a comma-separated string.
-      Capabilities: (capabilities && capabilities.join(',')) || undefined,
+      // None evaluates to empty string which is falsey and results in undefined
+      Capabilities: capabilities && capabilities.toString() || undefined,
```

Comment:
Oooh, you're relying on the fact that `&&` binds stronger than `||` - that is pure evil :). Mind adding some parenthesis around the 'and' expression?